### ISSUE TITLE
Support for Atlantic Calypso With CosyTouch

### ISSUE THREAD

**Issue by ioniks** [gduteil/cozytouch#5]
Hi,

If i can help for support Atlantic Calypso Connecté with CosyTouch.

;)

**Comment by gduteil** [comment]
Hi,

Is the device visible in Cozytouch application ? If it is but not detected by the integration it could mean it's using a different API. Does Overkiz detect anything ?

**Comment by ioniks** [comment]
Hi,

Yes this device is visible in cozytouch App and i can control it.
OverKiz detect it, i can use it (with many bug) with it.

**Comment by ioniks** [comment]
This year and the product is currently sale.

How i can test if it's compatible with the new api ?

**Comment by gduteil** [comment]
Do you see anything in the logs ?

We can also check what the API returns if you are comfortable modifying the integration script, I can guide you

**Comment by ioniks** [comment]
Nothing in the log.

Yes i can modify the integration script for test.

**Comment by gduteil** [comment]
The scripts are un /homeassistant/custom_components/cozytouch

You can try to add in validate_input function from config_flow.py the following line between "hub = Hub(..." and "result = await..." 😀 

hub.set_dump_json(True)

Save, restart HA and try again, it should dump "/homeassistant/Cozytouch.json"

You can post the file, make sure you replace private infos (address, serial number, ...) by XXXXX

**Comment by ioniks** [comment]
This is the result 
[Cozytouch.json](https://github.com/gduteil/cozytouch/files/14955252/Cozytouch.json)

**Comment by gduteil** [comment]
Yes and this is surprising, the gateway is present but no devices are attached... 

From what I saw, Atlantic devices sold in the past 2 or 3 years are using the new API, oldest ones use the Overkiz API, some sold during the transition are sometimes using both.

Is the gateway integrated in your device or is it external ?

**Comment by ioniks** [comment]
The gateway is external

**Comment by gduteil** [comment]
What is the model ?

**Comment by ioniks** [comment]
this 
[https://static-or00.inbenta.com/2285b3e873772835772543d5c2df2aed0394a0cfb9a47a4ec1416c77a7363c2a/bridge-cozytouch-2022.png](url)

i buy this because of not ethernet on the new one

**Comment by gduteil** [comment]
I'm afraid the V1 is using Overkiz, V2 is using Cozytouch, I haven't seen firmware update for V1 to use new API

**Comment by ioniks** [comment]
So for use your module i have to buy V2 or i wait for a firmware update (how can never append)

**Comment by gduteil** [comment]
Yes, I have never looked at the old protocol, I could try to make the integration compatible, we can imagine you temporary your password to share it with me the time I analyze the protocol if you want.

I'm currently away, I could start checking this in 2 weeks.

**Comment by ioniks** [comment]
Yes i can share you my temporary password to do it.

Come to me when you can

**Comment by gduteil** [comment]
Ok, you can send me an email on Gmail with same username I have here so we can find better way to communicate (WhatsApp or other)

**Comment by ioniks** [comment]
Mail sent

**Comment by Jmayormayu** [comment]
Hi Guys,
Thank you for your post. I have exactly the same products and same problems like @ioniks .
Have you found a workaround or a solution to this ?
Thank you !

**Comment by gduteil** [comment]
Older version use Overkiz API which is too different from the new API, unfortunately I will not have the time to make this integration compatible with Overkiz.

**Comment by diadeluz** [comment]
Hi @gduteil and thank you for your work!  

In your comment [above]([url] https://github.com/gduteil/cozytouch/issues/5#issuecomment-2051220575(url)) you are saying that the Cozytouch Bridge V2 does use a “new” API, also used by your integration (as opposed to the “old” Overkiz-based API used by Cozytouch Bridge V1).

So, **I am wondering whether your integration would be able to control my Thermor Ovation 3 radiators if I got rid of the Cozytouch V1 bridge and bought a V2 instead**.

Could you please let me know what you think? That would be awesome as I have had more than enough of the misbehaving Overkiz integration.

But then again, maybe it’s not only the bridge firmware but also the radiator firmware that would force me to stay with Overkiz? Or is there yet again another bridge I could buy like e.g. Solférino Tahoma?

Your views would be much appreciated :)
